src/plist.c

```diff
@@ -67,6 +67,11 @@ Int             GrowPlist (
     UInt                plen;           /* new physical length             */
     UInt                good;           /* good new physical length        */
 
+    if (!UINT_FITS_IN_INTOBJ(need)) {
+        ErrorMayQuit("GrowPlist: List size too large", 0, 0);
+    }
```

Comment:
This check is actually not enough: A few lines below, we also "round" up the current capacity of the list, i.e.
```C
    good = 5 * CAPACITY_PLIST(list) / 4 + 4;
```
So if the list was close to the limit before, then `good` will exceed the limit -- and then, even if `need` is below the limit, we end up using `good` (which is larger than `need`) as the new list length.

I think the correct solution then is to first check here; then compute `good`, and change 
```C
    if ( need < good ) { plen = good; }
    else               { plen = need; }
```
to something like
```
    if ( need < good && UINT_FITS_IN_INTOBJ(good) ) { plen = good; }
    else               { plen = need; }
```